pkg/platform/abstract/platform.go

```diff
@@ -516,22 +539,45 @@ func (ap *Platform) prettifyProcessorLogLine(log []byte) (string, string, error)
 
 	logLevel := strings.ToUpper(*logStruct.Level)[0]
 
-	messageAndArgs := ap.getMessageAndArgs(*logStruct.Message, logStruct.More, log)
+	// if worker ID wasn't explicitly given as an arg, infer worker ID from logger name
+	if workerID == "" {
+		workerID = ap.inferWorkerID(logStruct.Name)
+	}
+
+	messageAndArgs := ap.getMessageAndArgs(*logStruct.Message, logStruct.More, log, workerID)
 
 	res := fmt.Sprintf("[%s] (%c) %s",
 		parsedTime.Format("15:04:05.000"),
 		logLevel,
 		messageAndArgs)
 
 	briefLogLine := ""
-	if ap.shouldAddToBriefErrorsMessage(logLevel, *logStruct.Message) {
+	if ap.shouldAddToBriefErrorsMessage(logLevel, *logStruct.Message, workerID) {
 		briefLogLine = messageAndArgs
 	}
 
 	return res, briefLogLine, nil
 }
 
-func (ap *Platform) getMessageAndArgs(message string, args *string, log []byte) string {
+// get the worker ID from the logger name, for example:
+// "processor.http.w5.python.logger" -> 5
+func (ap *Platform) inferWorkerID(loggerName *string) string {
+	if loggerName == nil {
+		return ""
+	}
+
+	// if the logger name is the following pattern, extract the worker ID from it
+	processorRe := regexp.MustCompile(`^processor\..*\.w[0-9]+\..*`)
+	if processorRe.MatchString(*loggerName) {
+		splitName := strings.Split(*loggerName, ".")
+		return splitName[2][1:]
+	}
+
+	return ""
+}
+
+func (ap *Platform) getMessageAndArgs(message string, args *string, log []byte, workerID string) string {
```

Comment:
also here, when you don't want/need to change string contents - provide by value to keep the value immutable